prefect_gcp/cloud_storage.py

```diff
@@ -468,6 +474,74 @@ def example_cloud_storage_copy_blob_flow():
     return dest_blob
 
 
+class DataFrameSerializationFormat(Enum):
+    """
+    An enumeration class to represent different file formats,
+    compression options for upload_from_dataframe
+
+    Attributes:
+        CSV: Representation for 'csv' file format with no compression
+            and its related content type and suffix.
+
+        CSV_GZIP: Representation for 'csv' file format with 'gzip' compression
+            and its related content type and suffix.
+
+        PARQUET: Representation for 'parquet' file format with no compression
+            and its related content type and suffix.
+
+        PARQUET_SNAPPY: Representation for 'parquet' file format
+            with 'snappy' compression and its related content type and suffix.
+
+        PARQUET_GZIP: Representation for 'parquet' file format
+            with 'gzip' compression and its related content type and suffix.
+    """
+
+    CSV = ("csv", None, "text/csv", ".csv")
+    CSV_GZIP = ("csv", "gzip", "application/x-gzip", ".csv.gz")
+    PARQUET = ("parquet", None, "application/octet-stream", ".parquet")
+    PARQUET_SNAPPY = (
+        "parquet",
+        "snappy",
+        "application/octet-stream",
+        ".parquet.snappy",
+    )
+    PARQUET_GZIP = ("parquet", "gzip", "application/octet-stream", ".parquet.gz")
+
+    @property
+    def format(self) -> str:
+        """The file format of the current instance."""
+        return self.value[0]
+
+    @property
+    def compression(self) -> Union[str, None]:
+        """The compression type of the current instance."""
+        return self.value[1]
+
+    @property
+    def content_type(self) -> str:
+        """The content type of the current instance."""
+        return self.value[2]
+
+    @property
+    def suffix(self) -> str:
+        """The suffix of the file format of the current instance."""
+        return self.value[3]
+
+    def fix_extension_with(self, gcs_blob: str) -> str:
+        """Fix the extension of a GCS blob.
+
+        Args:
+            gcs_blob: The path to the GCS blob to be modified.
+
+        Returns:
+            The modified path to the GCS blob with the new extension.
+        """
+        gcs_blob = Path(gcs_blob)
+        folder = gcs_blob.parent
+        filename = Path(gcs_blob.stem).with_suffix(self.suffix)
```

Comment:
One last finding; on Windows, if we use Path, I think it converts to `bucket_folder\file_path.csv.gz` which GCS won't accept so we have to use PurePosixPath.
```suggestion
        gcs_blob_path = PurePosixPath(gcs_blob)
        folder = gcs_blob_path.parent
        filename = gcs_blob_path.stem.with_suffix(self.suffix)
```

Reply:
Also the tests are failing probably because of the changes:
```

cls = <enum 'DataFrameSerializationFormat'>, name = 'PICKLE'

    def __getitem__(cls, name):
>       return cls._member_map_[name]
E       KeyError: 'PICKLE'

```